### PR TITLE
Alias not created for the tranlsated node.

### PR DESCRIPTION
**Pull request by joshirohit100**
So if path auto module is enabled, an alias is not created for the translated version node.

**Steps** -

- Should be a multilingual site (more than one or say two languages enabled).

- Enable path auto module

- Set the pattern for the product bundle.

`$data = [
    'title' => 'Test Product Arabic',
    'type' => 'acm_product',
    'langcode' => 'ar',
  ];
  $node = \Drupal::entityTypeManager()->getStorage('node')->create($data);
  $node->save();

  $node = \Drupal\node\Entity\Node::load($node->id());
  $node = $node->addTranslation('en');
  $node->get('title')->setValue('Test Product English');
  $node->save();`

Alias is not created for the English version of the node.

Looks like a core/pathauto specific issue (not sure if we fix here or there).

Reason is `if ($entity->path->pathauto != PathautoState::CREATE && empty($options['force'])) { return NULL; }` in `PathautoGenerator::updateEntityAlias()` where

### CODE REVIEW

**Comment by joshirohit100** [comment]
cc @sdelbosc

File: modules/sku/src/ProductManager.php [hunk 1]
```diff
@@ -393,6 +393,15 @@ public function synchronizeProducts(array $products = [], $storeId = '') {
         elseif ($node->hasTranslationChanges()) {
           $this->updateNodeTranslation($node, $product, $langcode);
         }
+
+        // We doing this because when the translation of node is created by
+        // addTranslation(), pathauto alias is not created for the translated
+        // version.
+        // @see https://www.drupal.org/project/pathauto/issues/2995829.
+        if (\Drupal::moduleHandler()->moduleExists('pathauto')) {
```
Comment (miromichalicka) [inline modules/sku/src/ProductManager.php]: please inject `\Drupal::moduleHandler()` through DI

File: modules/sku/src/ProductManager.php [hunk 2]
```diff
@@ -393,9 +403,17 @@ public function synchronizeProducts(array $products = [], $storeId = '') {
         elseif ($node->hasTranslationChanges()) {
           $this->updateNodeTranslation($node, $product, $langcode);
         }
+
+        // We doing this because when the translation of node is created by
+        // addTranslation(), pathauto alias is not created for the translated
+        // version.
+        // @see https://www.drupal.org/project/pathauto/issues/2995829.
+        if ($this->moduleHandler->moduleExists('pathauto')) {
```
Comment (joshirohit100) [inline modules/sku/src/ProductManager.php:411]: This is the actual fix

File: modules/sku/src/ProductManager.php (hunk 1 above)
Comment (joshirohit100) [reply modules/sku/src/ProductManager.php]: PR updated! Thanks!

**Comment by joshirohit100** [comment]
As we are injecting the **module_handler** service, I have also updated the other usage of **\Drupal::moduleHandler()** in the class.

**Pull request merged**